test/openst/gnosisSafeModules/delayedRecoveryModule/resetRecoveryOwner.js

```diff
@@ -0,0 +1,114 @@
+    await openst.setupOpenst(
+      mockAddresses.organization,
+      mockAddresses.utilityToken,
+    )
+
+    const userWallet = {
+      owners: [mockAddresses.owner],
+      threshold: 1, // number of required confirmations
+      recoveryOwner: connection.auxiliaryAccount, // account that signs resetRecoveryOwner request
+      recoveryController: connection.auxiliaryAccount, // account that relays signed request
+      recoveryBlockDelay: 1, // not relevant to this test, must be greater than 0
+      sessionKeys: [], // not required
+      sessionKeySpendingLimits: [], // not required
+      sessionKeyExpirationHeights: [], // not required
+    }
+
+    // Setup user helper
+    //  adapted from ../../../../src/bin/create_user.js
+    const userHelper = new Helpers.User(
+      chainConfig.openst.tokenHolderMasterCopy,
+      chainConfig.openst.gnosisSafeMasterCopy,
+      chainConfig.openst.recoveryMasterCopy,
+      chainConfig.openst.createAndAddModules,
+      mockAddresses.utilityToken,
+      chainConfig.openst.tokenRules,
+      chainConfig.openst.userWalletFactory,
+      chainConfig.openst.proxyFactory,
+      web3,
+    );
+
+    // Create user wallet
+    //  deploys proxy contracts
+    //  adapted from ../../../../src/bin/create_user.js
+    const response = await userHelper.createUserWallet(
+      userWallet.owners,
+      userWallet.threshold,
+      userWallet.recoveryOwner.address,
+      userWallet.recoveryController.address,
+      userWallet.recoveryBlockDelay,
+      userWallet.sessionKeys,
+      userWallet.sessionKeySpendingLimits,
+      userWallet.sessionKeyExpirationHeights,
+      openst.auxiliary.txOptions,
+    );
+
+    // Instantiate DelayedRecoveryModule
+    const { returnValues } = response.events.UserWalletCreated;
+    const gnosisSafeProxy = returnValues._gnosisSafeProxy;
+    const gnosisSafe = new ContractInteract.GnosisSafe(web3, gnosisSafeProxy);
+    const modules = await gnosisSafe.getModules();
+    const recoveryProxy = modules[0];
+    const delayedRecoveryModule = new ContractInteract.Recovery(web3, recoveryProxy);
+
+    // Confirm recoveryOwner is as expected
+    assert.strictEqual(
+      userWallet.recoveryOwner.address,
+      await delayedRecoveryModule.recoveryOwner(),
+    )
+
+    // Calculate recovery data
+    const resetRecoveryOwnerData = delayedRecoveryModule.resetRecoveryOwnerData(
+      userWallet.recoveryOwner.address,
+      mockAddresses.newRecoveryOwner,
+    );
+
+    // Sign recovery data
+    //  openst.js extends Account to include signEIP712TypedData (via mosaic.js)
+    const signedRecoveryData = await userWallet
+      .recoveryOwner
+      .signEIP712TypedData(resetRecoveryOwnerData);
+
+    // Reset recoveryOwner
+    await delayedRecoveryModule.resetRecoveryOwner(
+      mockAddresses.newRecoveryOwner,
+      signedRecoveryData.r,
+      signedRecoveryData.s,
+      signedRecoveryData.v,
+      openst.auxiliary.txOptions,
```

Comment:
The call should come from recovery controller address. Assuming it's the same here, however, in test we should make  sure that if it's not it's still passing.

Reply:
The recovery controller address is the `from` in `txOptions`, but that is not immediately clear—will add a comment.

I don't think I understand the rest of the statement (three "it's" that I think may not all mean the same thing 😅 ).